tensorrt_llm/_torch/attention_backend/interface.py

```diff
@@ -160,7 +160,16 @@ def seq_lens(self, value: Optional[torch.Tensor]):
         # The model executor sets seq_lens to None initially.
         if self._seq_lens is not None:
             self._seq_lens = self._seq_lens.pin_memory()
-            self._seq_lens_cuda = self._seq_lens.cuda(non_blocking=True)
+
+            if self.is_cuda_graph and self._seq_lens_cuda is not None:
+                # Very important: do not reallocate if we are using CUDA graphs.
+                # This copy is safe because the batch size is guaranteed to not
+                # change in the CUDA graph case. The seqlens can change if we
+                # are doing spec decode.
+                self._seq_lens_cuda.copy_(self._seq_lens)
+            else:
+                self._seq_lens_cuda = self._seq_lens.cuda(non_blocking=True)
```

Comment:
This part is a bit hard to follow. When will this branch get hit?

How are these two lines different:
```
self._seq_lens_cuda.copy_(self._seq_lens)
self._seq_lens_cuda = self._seq_lens.cuda(non_blocking=True)
```
On first look, they're both copying from `_seq_lens` to `_seq_lens_cuda`